CDS/src/org/icpc/tools/cds/video/containers/HLSHandler.java

```diff
@@ -0,0 +1,270 @@
+package org.icpc.tools.cds.video.containers;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.icpc.tools.cds.video.VideoServingHandler;
+import org.icpc.tools.cds.video.VideoStream;
+import org.icpc.tools.cds.video.VideoStreamHandler.IStreamListener;
+import org.icpc.tools.contest.Trace;
+import org.icpc.tools.contest.model.feed.HTTPSSecurity;
+
+/**
+ * HLS handler. https://localhost:8443/stream/11?_HLS_msn=10&_HLS_part=3&_HLS_skip=YES
+ * https://localhost:8443/stream/11?_HLS_msn=10&_HLS_part=4&_HLS_skip=YES
+ */
+// TODO - cache HLS m3u8
+// TODO - handle HLS query params
```

Comment:
These are new for me, seems they are introduced when they introduced low latency HLS: https://developer.apple.com/documentation/http-live-streaming/enabling-low-latency-http-live-streaming-hls#Add-Low-Latency-HLS-Delivery-Directives

Reply:
Yes. Looks like It essentially gives you a way to block until content is available, so CDS will need to be intelligent about this too.